# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/WasmBCE.cpp
# Commit: 90fa382b5a20
# Full Hash: 90fa382b5a203d2e28bfa69f29ab8edee150b77b
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 8
# Description:
#   Bug 1642412 - wasm: Optimize ion table.get/set/size. r=lth
#   
#   This commit adds ion fast paths for table.get/set when the table
#   representation is anyref. An unconditional fast path is added for
#   table.size.
# ==============================================================================

diff -r 40f861f62a0f -r 90fa382b5a20 js/src/jit/WasmBCE.cpp
--- a/js/src/jit/WasmBCE.cpp	Wed Dec 22 17:36:29 2021 +0000
+++ b/js/src/jit/WasmBCE.cpp	Wed Dec 22 17:36:30 2021 +0000
@@ -45,7 +45,13 @@
           MWasmBoundsCheck* bc = def->toWasmBoundsCheck();
           MDefinition* addr = bc->index();
 
-          // Eliminate constant-address bounds checks to addresses below
+          // We only support bounds check elimination on wasm memory, not
+          // tables. See bug 1625891.
+          if (!bc->isMemory()) {
+            continue;
+          }
+
+          // Eliminate constant-address memory bounds checks to addresses below
           // the heap minimum.
           //
           // The payload of the MConstant will be Double if the constant